SA3/qa_web/templates/qa_web/answerspage.html

```diff
@@ -68,36 +68,71 @@ <h3 class="good_answer">{{currentQuestion.owner}} has selected this as the best
       <a id='downvote_{{ bestAnswer.id }}_answer'><button>Downvote</button></a>
-      <hr>
+      {% for comment in comments %}
+        {% if comment.answer.id == bestAnswer.id %}
+            <p>{{ comment.content }} (Comment by: {{ comment.owner }}. Date posted: {{ comment.creation_date }})
+                 <a id='upvote_{{ comment.id }}_comment'><button>Upvote</button></a>
+                 <span id='score_{{ comment.id }}_comment'>{{ comment.score }}</span>
+                 <a id='downvote_{{ comment.id }}_comment'><button>Downvote</button></a>
+            </p>
+        {% endif %}
+      {% endfor %}
+      {% if user.is_authenticated %}
+        <p>Got a comment? Why not share it here!</p>
+        <form id="comment_form_{{ bestAnswer.id }}" method="post">
+          {% csrf_token %}
+            <textarea id="wmd-input" name="content" class="wmd-input"></textarea></br>
+            <input type="submit" name="comment_form_{{ bestAnswer.id }}" value="Submit your comment"/>
+         </form>
+      {% endif %}
       <h3>Other answers:</h3>
+      <hr>
   {% endif %}
   {% for answer in answers %}
       <form id="select{{ answer.id }}_form" method="post">
           {% csrf_token %}
           <p>{{ answer.content }}</p>
           <!--Allow one to select best answers if user is logged in, is the user who asked the question and a best answer doesn't exist already-->
-          {% if user.is_authenticated and user == currentQuestion.owner and not bestAnswer %} 
-             <input type = "submit" name="select_{{ answer.id }}" value="Select as Best Answer"></input>
+          {% if user.is_authenticated and user == currentQuestion.owner and not bestAnswer %}
+             <input type = "submit" name="select_{{ answer.id }}" value="Select as Best Answer">
           {% endif %}
           <h6>Answer by {{ answer.owner }} on {{ answer.creation_date }}</h6>
       </form>
       <a id='upvote_{{ answer.id }}_answer'><button>Upvote</button></a>
       <span id='score_{{ answer.id }}_answer'>{{ answer.score }}</span>
       <a id='downvote_{{ answer.id }}_answer'><button>Downvote</button></a>
+      {% for comment in comments %}
+        {% if comment.answer.id == answer.id %}
+            <p>{{ comment.content }} (Comment by: {{ comment.owner }}. Date posted: {{ comment.creation_date }})
+                 <a id='upvote_{{ comment.id }}_comment'><button>Upvote</button></a>
+                 <span id='score_{{ comment.id }}_comment'>{{ comment.score }}</span>
+                 <a id='downvote_{{ comment.id }}_comment'><button>Downvote</button></a>
+            </p>
+        {% endif %}
+      {% endfor %}
+      {% if user.is_authenticated %}
+        <p>Got a comment? Why not share it here!</p>
+        <form id="comment_form_{{answer.id}}" method="post">
+          {% csrf_token %}
+            <textarea id="wmd-input" name="content" class="wmd-input"></textarea></br>
```

Comment:
I think this text area should either be shown on event trigger or appear as a smaller box. Right now the boxes obstruct the reading flow of the thread. This can be fixed in another issue.